pkg/ctrl/config.go

```diff
@@ -0,0 +1,33 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ctrl
+
+import "k8s.io/client-go/rest"
+
+// Config provides a rest.Config that can be used to identify and authenticate with a Kubernetes cluster.
+type Config interface {
+	Config() rest.Config
+}
+
+var _ Config = ConfigFn(func() rest.Config { return rest.Config{} })
+
+// ConfigFn implements Config
```

Comment:
Maybe use the usual `_ Config` trick?

Reply:
Are you suggesting something different from what is on line 26?